source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -432,13 +446,98 @@ void SplitKeysSumResultRequest::onChildResponse(Common::Redis::RespValuePtr&& va
   }
 }
 
+SplitRequestPtr TransactionRequest::create(Router& router,
+                                           Common::Redis::RespValuePtr&& incoming_request,
+                                           SplitCallbacks& callbacks, CommandStats& command_stats,
+                                           TimeSource& time_source, bool delay_command_latency) {
+  Common::Redis::Client::Transaction& transaction = callbacks.transaction();
+  std::string to_lower_string = absl::AsciiStrToLower(incoming_request->asArray()[0].asString());
+
+  // Within transactions we only support simple commands.
+  // So if this is not a transaction command or a simple command, it is an error.
+  if (Common::Redis::SupportedCommands::transactionCommands().count(to_lower_string) == 0 &&
+      Common::Redis::SupportedCommands::simpleCommands().count(to_lower_string) == 0) {
+    callbacks.onResponse(Common::Redis::Utility::makeError(
+        fmt::format("'{}' command is not supported within transaction",
+                    incoming_request->asArray()[0].asString())));
+    return nullptr;
+  }
+
+  // Start transaction on MULTI, and stop on EXEC/DISCARD.
+  if (to_lower_string == "multi") {
+    // Check for nested MULTI commands.
+    if (transaction.active_) {
+      callbacks.onResponse(
+          Common::Redis::Utility::makeError(fmt::format("MULTI calls can not be nested")));
+      return nullptr;
+    }
+    transaction.start();
+    // Respond to MULTI locally.
+    localResponse(callbacks, "OK");
+    return nullptr;
+
+  } else if (to_lower_string == "exec" || to_lower_string == "discard") {
+    // Handle the case where we don't have an open transaction.
+    if (transaction.active_ == false) {
+      callbacks.onResponse(Common::Redis::Utility::makeError(
+          fmt::format("{} without MULTI", absl::AsciiStrToUpper(to_lower_string))));
+      return nullptr;
+    }
+
+    // Handle the case where the transaction is empty.
+    if (transaction.key_.empty()) {
+      if (to_lower_string == "exec") {
+        localResponse(callbacks, "(empty array)");
```

Comment:
I think "(empty array)" is a message printed from redis-cli. The actual RESP message returned after MULTI and EXEC is "*0\r\n":

```
> nc -q 1 localhost 6379
MULTI
+OK
EXEC
*0
```